tests/infra/network.py

```diff
@@ -1260,6 +1260,12 @@ def wait_for_snapshots_to_be_committed(src_dir, list_src_dir_func, timeout=20):
                         )
                         return True
 
+                if time.time() > end_time:
+                    LOG.error(
+                        f"Could not find committed snapshot for seqno {target_seqno} after {timeout:.2f}s in {src_dir}: {list_src_dir_func(src_dir)}"
+                    )
+                    return False
+
```

Comment:
Restructuring this while loop to check the condition here, rather than at the start. The problem with the other form is that when the node is running slow, the calls to `update_state_digest` and `wait_for_commit` can take an extremely long time, and once they finish we don't actually check if a snapshot has been produced in the interim. This form ensures we always checking for a snapshot _after_ manually inducing one, even if that induction takes a long time and only permits us a single iteration.